Classes/ResourceImportAspect.php

```diff
@@ -126,17 +121,15 @@ public function importOnGetPublicPersistentResourceUri(JoinPointInterface $joinP
 
     private function importRemoteResource(PersistentResource $resource, WritableStorageInterface $storage): void
     {
-        if ($storage instanceof WritableFileSystemStorage) {
-            $pathAndFilename = $this->getStoragePathAndFilename($resource->getSha1(), $storage);
+        if ($storage instanceof WritableStorageInterface) {
             $content = $this->getRemoteResource($resource, $storage);
             if ($content === false) {
                 $this->logger->notice(sprintf('Could not fetch resource data for "%s".', $resource->getSha1()), LogEnvironment::fromMethodName(__METHOD__));
                 return;
             }
 
-            Files::createDirectoryRecursively(dirname($pathAndFilename));
-            file_put_contents($pathAndFilename, $content);
-            $this->logger->notice(sprintf('Stored resource data at "%s".', $pathAndFilename), LogEnvironment::fromMethodName(__METHOD__));
+            $storage->importResourceFromContent($content, $resource->getCollectionName());
```

Comment:
This should work as long as the storage doesn't do anything funky to the content so that it ends up having a different hash than expected, but I guess it's a fair risk to take.